launcher/signature-discovery/client/client.go

```diff
@@ -0,0 +1,63 @@
+// Package client contains functionalities to discover container image signatures.
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/images"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const signatureTagSuffix = "sig"
+
+// Client is a wrapper of containerd.Client to interact with signed image manifest.
+type Client struct {
+	cdClient      *containerd.Client
+	OriginalImage containerd.Image
+	RemoteOpts    []containerd.RemoteOpt
+}
+
+// New creates a new client.
+func New(cdClient *containerd.Client, originalImage containerd.Image, opts ...containerd.RemoteOpt) *Client {
+	return &Client{
+		cdClient:      cdClient,
+		OriginalImage: originalImage,
+		RemoteOpts:    opts,
+	}
+}
+
+// FetchSignedImageManifest fetches a signed image manifest using a tag-based discovery mechanism.
+func (c *Client) FetchSignedImageManifest(ctx context.Context, targetRepository string) (v1.Manifest, error) {
+	image, err := c.pullTargetImage(ctx, targetRepository)
+	if err != nil {
+		return v1.Manifest{}, err
+	}
+	return getManifest(ctx, image)
+}
+
+func (c *Client) pullTargetImage(ctx context.Context, targetRepository string) (containerd.Image, error) {
+	targetImageRef := fmt.Sprint(targetRepository, ":", formatSigTag(c.OriginalImage))
+	image, err := c.cdClient.Pull(ctx, targetImageRef, c.RemoteOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("[signature-discovery]: cannot pull the image [%s] from taregetRepository [%s]: %w", targetImageRef, targetRepository, err)
```

Comment:
We generally don't have prefixes for components in our logs. To keep our logging consistent, let's do something like: `"cannot pull signature object '%s' from target repository '%s': %w"`